vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -613,4 +620,87 @@ private void fireSizeEvent() {
     private void identifierProviderChanged(IdentifierProvider<T> identifierProvider) {
         keyMapper.setIdentifierGetter(identifierProvider);
     }
+
+    @SuppressWarnings("rawtypes")
+    private Query getQuery() {
+        return new Query<>(0, Integer.MAX_VALUE, null,
+                InnerCheckboxGroupListDataView.getComponentSortComparator(this)
+                        .orElse(null),
+                InnerCheckboxGroupListDataView.getComponentFilter(this)
+                        .orElse(null));
+    }
+
+    private void removeFilteringAndSorting() {
+        InnerCheckboxGroupListDataView.removeComponentFilter(this);
+        InnerCheckboxGroupListDataView.removeComponentSortComparator(this);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static final class InnerCheckboxGroupListDataView
```

Comment:
This seems a bit excessive adding this weird class with a ton of code to maintain just to expose these one-liner (static!) delegating methods to the two methods above. For instance it would be easier if the above methods would be exposed instead in a place where they can access the static methods in the abstract list data view. There are probably multiple options for this

Reply:
Or maybe even moving the static methods in Flow side to be public but in a util class to avoid the boiler plate in the components and still make those "invisible" to the users ?

Reply:
Alright, adding one more util class on Flow side looks good. So, I'll create a `DataViewUtils` class in `com.vaadin.flow.data.provider` and move these static methods there and make them public. Then I use them in `AbstractListDataView` and in components

Reply:
Done.